src/services/ksSetting.ts

```diff
@@ -0,0 +1,61 @@
+import { ChainId } from '@kyberswap/ks-sdk-core'
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+
+import { KS_SETTING_API } from 'constants/env'
+
+export type KyberswapConfigurationResponse = {
+  data: {
+    prochart: boolean
+    rpc: string
+    'block-subgraph': string
+    'classic-subgraph': string
+    'elastic-subgraph': string
+  }
+}
+
+type KyberswapGlobalConfigurationResponse = {
+  data: {
+    banner: {
+      id: string
+      name: string
+      start: string
+      end: string
+      link: string
+      img: {
+        modal: string
+        larger: string
+        [media: string]: string
+      }
+    }
+  }
+}
+
+const ksSettingApi = createApi({
+  reducerPath: 'ksSettingConfigurationApi',
+  baseQuery: fetchBaseQuery({
+    baseUrl: `${KS_SETTING_API}/v1`,
+  }),
+  endpoints: builder => ({
+    getKyberswapConfiguration: builder.query<KyberswapConfigurationResponse, { chainId: ChainId }>({
+      query: ({ chainId }) => ({
+        url: '/configurations/fetch',
+        params: {
+          serviceCode: `kyberswap-${chainId}`,
+        },
+      }),
+    }),
+    getKyberswapGlobalConfiguration: builder.query<KyberswapGlobalConfigurationResponse, undefined>({
```

Comment:
undefined => void

Reply:
is there any different between them?

Reply:
undefined is value
void is type

Reply:
undefined can be used as type too right?

Reply:
yes, It's is my suggestion, it's up to u